### DISCUSSION TITLE
Can I use GPU with neural-prophet?

### DISCUSSION THREAD

**Discussion by merchen911** [ourownstory/neural_prophet#408] [Feature Requests & Feedback]
In the code, even though based on the torch, I cant use GPU.

I am using it for the version currently being distributed after personally modifying it.

Hope to know when the neural prophet will support GPU acceleration.

**Comment by itay-rubin** [comment]
someone?

**Comment by wangzhongqing** [comment]
Can you save the model file locally? Can you tell me, thank you？

**Reply by ourownstory** [reply]
currently there is no utility, but you can do it like this:
https://pytorch.org/tutorials/beginner/saving_loading_models.html

Tracked in #27

**Reply by wangzhongqing** [reply]
Thx,i will try it.

**Comment by rishabh1212** [comment]
@ourownstory is it possible to use GPU with neural-prophet? or planned for future release??

**Comment by ourownstory** [comment]
@merchen911 @rishabh1212 
Currently, there is no GPU support. We have planned it, but it's likely not soon, as we would first need to achieve better parallelization of the model for this to make a big difference.

However, if you think it may already help you, please feel free to contribute a PR adding GPU support!

**Reply by laoliang95** [reply]
I would like to ask how NeuralProphet uses GPU

**Comment by noxan** [comment]
@merchen911  @itay-rubin @wangzhongqing @rishabh1212 We added gpu support based on lightning with the latest release candidate `0.5.0rc1` - happy to hear your feedback and let us know if it works for you - see [release notes](https://github.com/ourownstory/neural_prophet/releases/tag/0.5.0rc1) or the [related pull request](https://github.com/ourownstory/neural_prophet/pull/961).

You can either install from source or via pypi with our pre-release version (you have to manually specify exactly `0.5.0rc1` or use the `--pre` flag.

**Reply by hjmok** [reply]
hi @noxan , I've installed neuralprophet==0.5.0rc1, and run the following code 

`trainer_config = {"accelerator":"gpu"}`

```
m = NeuralProphet(
    weekly_seasonality=6,
    daily_seasonality=10,
    trend_reg=1,
    learning_rate=0.01,
    trainer_config=trainer_config
)
```

`df_train, df_test = m.split_df(sf_load_df, freq="H", valid_p=1.0 / 12)`

`metrics = m.fit(df_train, freq="H", validation_df=df_test, progress="plot")`

However, I still run into this error
`RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!`

I have checked that my torch is connected and currently using my NVIDIA GeForce GTX 1650 Ti from this command
`print(torch.cuda.get_device_name(0))`

Any suggestions?

**Reply by noxan** [reply]
Hi @hjmok,

@karl-richter is already looking into this, see https://github.com/ourownstory/neural_prophet/issues/1002 and there is meanwhile a fix in review with #1023 - feel free to check out that version and let us know in the issue if it resolves your issue :)
